### PR TITLE
[15.0][MIG] hr_attendance_report_theoretical_time: Migration to 15.0

### PR DESCRIPTION
**Pull request by victoralmau**
Migration to 15.0

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT36461

### CODE REVIEW

**Comment by ghost** [comment]
Hi,
A employee starts working since 01/01/22 with a calendar of 40h/week. Today is August 04th.
You change the working calendar on employee to 35h/week.
You want to change the records since july 01st. From january to june on fridays it computes 7 theorical hours instead 8.
2022-08-04 13_27_03-Odoo - Standard 40 hours_week

The rest seems OK

**Comment by pedrobaeza** [comment]
Is this something specific of the migration? What happens on previous versions?

**Comment by victoralmau** [comment]
> 

After making this change, it is necessary to re-compute the theoretical hours from the corresponding menu (selecting employee + start date and end date).

Anyway, I think this is a question rather than a migration problem.

**Comment by ghost** [comment]
@victoralmau what I have said is happening after recompute theoretical attendances (selecting employee + start date and end date).
You can try on runboat

**Comment by pedrobaeza** [comment]
I repeat: is this something specific of this version migration or something that is happening in other versions?

**Comment by ghost** [comment]
On V14 it happens the same. But it happens only if the employee has attendances registered. If the employee doesn't, doing the same test, all registered have been changed to 7 hours. Even january to june.
But it seems that is not a migration issue.

**Review by ghost — APPROVED** [review]

**Comment by pedrobaeza** [comment]
OK, then please put an issue with such steps to reproduce the problem and the affected versions to be handled isolated from this migration.

File: hr_attendance_report_theoretical_time/reports/hr_attendance_report.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+
+class HRAttendanceReport(models.Model):
+    _inherit = "hr.attendance.report"
+
+    theoretical_hours = fields.Float(readonly=True)
+
+    # pylint: disable=sql-injection,missing-return
+    def init(self):
+        """Inject parts in the query with this hack, fetching the query and
```
Comment (pedrobaeza) [inline hr_attendance_report_theoretical_time/reports/hr_attendance_report.py]: Which parts you are injecting? Which is the goal of them?

File: hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report.py [hunk 2]
```diff
@@ -195,6 +195,8 @@ def _theoretical_hours(self, employee, date):
                 ("holiday_id.holiday_status_id.include_in_theoretical", "=", False),
             ],
         )[
+            employee.id
+        ][
             "hours"
         ]
```
Comment (pedrobaeza) [inline hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report.py]: Can't this be joined for better layout? (black has evolved meanwhile).

**Review by pedrobaeza — APPROVED** [review]

**Review by chienandalu — APPROVED** [review]
Tested. I'm just wondering if the overtime features should be considered....

**Comment by pedrobaeza** [comment]
/ocabot merge nobump 

what do you mean, @chienandalu ?

**Pull request merged**